examples/snake/main.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+package main
+
+import (
+	"fmt"
+	"image/color"
+	"log"
+	"math/rand"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	"github.com/hajimehoshi/ebiten/inpututil"
+)
+
+const (
+	screenWidth  = 640
+	screenHeight = 480
+	gridSize     = 10
+	xNumInScreen = screenWidth / gridSize
+	yNumInScreen = screenHeight / gridSize
+)
+
+const (
+	dirNone = iota
+	dirLeft
+	dirRight
+	dirDown
+	dirUp
+)
+
+type Position struct {
+	X int
+	Y int
+}
+
+type Game struct {
+	moveDirection int
+	snakeBody     []Position
+	apple         Position
+	timer         int
+	moveTime      int
+	score         int
+	bestScore     int
+	level         int
+}
+
+func (g *Game) collidesWithApple() bool {
+	return g.snakeBody[0].X == g.apple.X &&
+		g.snakeBody[0].Y == g.apple.Y
+}
+
+func (g *Game) collidesWithSelf() bool {
+	for _, v := range g.snakeBody[1:] {
+		if g.snakeBody[0].X == v.X &&
+			g.snakeBody[0].Y == v.Y {
+			return true
+		}
+	}
+	return false
+}
+
+func (g *Game) collidesWithWall() bool {
+	return g.snakeBody[0].X < 0 ||
+		g.snakeBody[0].Y < 0 ||
+		g.snakeBody[0].X >= xNumInScreen ||
+		g.snakeBody[0].Y >= yNumInScreen
+}
+
+func (g *Game) needsToMoveSnake() bool {
+	return g.timer%g.moveTime == 0
+}
+
+func (g *Game) reset() {
+	g.apple.X = 3 * gridSize
+	g.apple.Y = 3 * gridSize
+	g.moveTime = 4
+	g.snakeBody = g.snakeBody[:1]
+	g.snakeBody[0].X = xNumInScreen / 2
+	g.snakeBody[0].Y = yNumInScreen / 2
+	g.score = 0
+	g.level = 1
+	g.moveDirection = dirNone
+}
+
+func (g *Game) Update(screen *ebiten.Image) error {
+
```

Comment:
I think this is the last one request: Remove this empty line.

Reply:
done